### PR TITLE
[MIG] stock_orderpoint_generator: Migration to 12.0

### PR DESCRIPTION
**Pull request by chienandalu**
Standar migration

cc @Tecnativa TT20139

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: stock_orderpoint_generator/models/orderpoint_template.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        string='Products',
+        help="A reordering rule will be automatically created by the "
+             "scheduled action for every product in this list."
+    )
+    auto_last_generation = fields.Datetime(string='Last Automatic Generation')
+
+    def _template_fields_to_discard(self):
+        """In order to create every orderpoint we should pop this template
+           customization fields """
+        return [
```
Comment (Tardo) [inline stock_orderpoint_generator/models/orderpoint_template.py:84]: I see that this is used for "search" purposes... Use 'set' can increase the performance.

File: stock_orderpoint_generator/models/product.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        # default the compute the stock value anyway to default the value
+        # for products with no moves for the given period
+        initial_stock = {}
+        initial_stock = self.with_context(
+            location=location)._compute_quantities_dict(
+                False, False, False, to_date=from_date or to_date)
+        product_moves_dict = {}
+        for move in moves:
+            product_moves_dict.setdefault(move['product_id'][0], {})
+            product_moves_dict[move['product_id'][0]].update({
```
Comment (Tardo) [inline stock_orderpoint_generator/models/product.py:84]: Why don't you do it on the previous line?

**Review by Tardo — APPROVED** [review]
LGTM only non-blocking suggestions

File: stock_orderpoint_generator/models/product.py (hunk 2 above)
Comment (chienandalu) [reply stock_orderpoint_generator/models/product.py:84]: There could be no key yet

File: stock_orderpoint_generator/models/orderpoint_template.py (hunk 1 above)
Comment (chienandalu) [reply stock_orderpoint_generator/models/orderpoint_template.py:84]: Well, this method it's just to discard the fields not present in the original model used for templating purposes.

**Comment by pedrobaeza** [comment]
As the questions have been answered, let's merge it:

/ocabot merge

File: stock_orderpoint_generator/models/orderpoint_template.py [hunk 3]
```diff
@@ -36,12 +36,8 @@ class OrderpointTemplate(models.Model):
         help="Auto compute minimum quantity "
              "per product for a given a date range",
     )
-    auto_min_date_start = fields.Datetime(
-        string="From",
-    )
-    auto_min_date_end = fields.Datetime(
-        string="To",
-    )
+    auto_min_date_start = fields.Datetime()
```
Comment (rousseldenis) [inline stock_orderpoint_generator/models/orderpoint_template.py:39]: @chienandalu Why removing string ?

Comment (pedrobaeza) [reply stock_orderpoint_generator/models/orderpoint_template.py:39]: He has removed them in favor of the automatic ones for avoiding the warning of "duplicated labels", which in fact is more correct (for searches for example). They are put however on the view instead, for better display.

Comment (rousseldenis) [reply stock_orderpoint_generator/models/orderpoint_template.py:39]: Ok, saw in form, but for exports or ... the name will be Auto Min Date Start which could lead to inconsistencies

Comment (pedrobaeza) [reply stock_orderpoint_generator/models/orderpoint_template.py:39]: Well, on export you don't have context, so better to have full name, although it's not the same as you see in screen.

Comment (chienandalu) [reply stock_orderpoint_generator/models/orderpoint_template.py:39]: Yes, that was the exact reason :)

**Pull request merged**